tools/cabana/streams/abstractstream.h

```diff
@@ -87,5 +87,16 @@ class AbstractStream : public QObject {
   uint64_t last_event_ts = 0;
 };
 
+class AbstractOpenStreamWidget : public QWidget {
```

Comment:
You could also add a static method to `AbstractStream` to build the wdiget. E.g. `static AbstractOpenStreamWidget* getOpenStreamWidget()`. Then the rest of the code only needs to know about a single stream class for each implementation.